### DISCUSSION TITLE
How to use IfcConvert, BlenderBIM and Ifc2CA from python ?

### DISCUSSION THREAD

**Discussion by colasnaudi** [IfcOpenShell/IfcOpenShell#4674] [General]
Hi, 
I need to convert an ifc file to gltf, json and svg. But i don't know how to use all the libraries inside ifcopenshell.
I have no problem using IfcConvert from a python file because IfcConvert exists but my question is how to do the same with BlenderBIM and Ifc2CA ?
(I work with python)

Thanks :)

**Comment by aothms** [comment]
I don't know a lot about code aster, but the module you're referring to is a python module so that should be trivial to integrate https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.7.0/src/ifc2ca/ifc2ca.py If there are things you encounter that can be documented better or introduce a more straight forward entry point for programmatic usage, feel free to send in a Pull Request.

For gltf you can either call IfcConvert as a subprocess. Or use the serializers directly in python (since fairly recently). https://github.com/IfcOpenShell/IfcOpenShell/blob/a2ee920a5fd60b39c2815d63f57a0f2ab727f137/src/ifcopenshell-python/docs/ifcopenshell-python/geometry_processing.rst#geometry-serialisation

For SVG. You can use IfcConvert/python serializer, but you can also look at https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.7.0/src/ifcopenshell-python/ifcopenshell/draw.py which does a bit more (most notably colouring the cells from the hidden line projections which would otherwise just be disjoint lines). This module is mostly in line with the blenderbim functionality, but there are also some differences that are not fully known to me.
